rust/signed_doc/src/builder/cbor_map.rs

```diff
@@ -0,0 +1,49 @@
+use std::collections::BTreeMap;
+
+use super::EncodeError;
+
+/// A map of CBOR encoded key-value pairs with **bytewise** lexicographic key ordering.
+#[derive(Debug, Default)]
+pub struct CborMap(BTreeMap<Vec<u8>, Vec<u8>>);
```

Comment:
CBOR map is not a BtreeMap or HashMap, it allows to have multiple values under the same key.
```suggestion
pub struct CborMap(Vec<Vec<u8>, Vec<u8>>);
```

Reply:
>it's not in the case of COSE, though. There should be unique keys in metadata by specification.

Sorry, you are right, just checked spec again.

> `BTreeMap` provides the deterministic order

Yes, agree but this order is out of our control, we cannot control the order of inserted items, which is not what we want here. We need to preserve how they were inserted

Reply:
>but we were taking about deterministic CBOR, and the key ordering is an integral part here.

Yes exactly and that means that I some cases I want to make a cbor map for example with the following values (because it was already defined by our spec, that the keys should be exactly in this order)
```
{
 "some_thing": 1,
 "something_2": 2
}
```
And `BTreeMap` could prevent me to do that, because it will apply its own ordering rule, different what I was wanted originally ("some_thing" key is first, "something_2" is second).

Reply:
Do we have such specs?
I'd really like to hear @stevenj's perspective here

Reply:
`indexmap` is fine, you could use it